libecole/include/ecole/environment/environment.hpp

```diff
@@ -153,16 +166,23 @@ class Environment {
 	 *      In such cases, a call to reset must be perform before continuing.
 	 */
 	template <typename... Args>
-	auto step(Action const& action, Args&&... args) -> std::tuple<Observation, ActionSet, Reward, bool, InformationMap> {
+	auto step(Action const& action, Args&&... args)
+		-> std::tuple<OptionalObservation, ActionSet, Reward, bool, InformationMap> {
 		if (!can_transition) {
 			throw Exception("Environment need to be reset.");
 		}
 		try {
 			auto const [done, action_set] = dynamics().step_dynamics(model(), action, std::forward<Args>(args)...);
 			can_transition = !done;
 
+			OptionalObservation observation;
+			if (!done) {
+				observation = observation_function().extract(model(), done);
+			} else {
+				observation = OptionalObservation{};
+			}
```

Comment:
Same suggestion here:
```suggestion
			auto observation =  OptionalObservation{};
			if (!done) {
				observation = observation_function().extract(model(), done);
			}
```